### PR TITLE
Ask users to provide a new google oauth credential if current one is invalid

### PR DESCRIPTION
**Pull request by FrankApiyo**
### Changes / Features implemented

- Delete invalid google OAuth credential and ask users to re-create them

### Steps taken to verify this change does what is intended
- Added a test case
### Side effects of implementing this change
- N/A

**Before submitting this PR for review, please make sure you have:**

  - [x] Included tests
  - [ ] Updated documentation

### CODE REVIEW

File: onadata/libs/utils/api_export_tools.py [hunk 1]
```diff
@@ -636,6 +637,11 @@ def _get_google_credential(request):
     elif request.session.get("access_token"):
         credential = Credentials(token=request.session["access_token"])
 
+    if credential and not credential.valid:
```
[the remark this answers is not in the document]
Comment (DavisRayM) [reply onadata/libs/utils/api_export_tools.py:640]: Could we perhaps check if it's valid then try to [refresh](https://google-auth.readthedocs.io/en/stable/reference/google.oauth2.credentials.html#google.oauth2.credentials.Credentials.refresh) the token manually and then clear it if the refresh did not successfully happen

[the remark this answers is not in the document]
Comment (FrankApiyo) [reply onadata/libs/utils/api_export_tools.py:643]: done!

[the remark this answers is not in the document]
Comment (FrankApiyo) [reply onadata/libs/utils/api_export_tools.py:643]: I copied this way of refreshing tokens from the documentation for request [here](https://google-auth.readthedocs.io/en/stable/reference/google.auth.transport.requests.html#google.auth.transport.requests.Request)

I also tried it out on my own token and it seems to work as expected

File: onadata/libs/utils/api_export_tools.py [hunk 2]
```diff
@@ -638,9 +641,11 @@ def _get_google_credential(request):
         credential = Credentials(token=request.session["access_token"])
 
     if credential and not credential.valid:
-        # We need the user to provide a new crential
-        storage.delete()
-        credential = None
+        try:
+            credential.refresh(Request())
```
Comment (DavisRayM) [inline onadata/libs/utils/api_export_tools.py:645]: Should we store the credential again once we've refreshed it ? _We may also need to somehow block other requests from refreshing the token while we refresh it in the future perhaps..._

**Review by DavisRayM — APPROVED** [review]
LGTM

**Comment by KipSigei** [comment]
@FrankApiyo Could we add a test case for refreshing credentials and check whether they are valid when the refresh is successful?

**Comment by FrankApiyo** [comment]
> @FrankApiyo Could we add a test case for refreshing credentials and assert whether they are valid or not expired when the refresh is successful?

This would be a bit tricky because we would need to provide an example of a valid google credential. And that would have to be attached to a google account. I also think such a test would be testing google oauth API and not our code, don't you think so?

**Review by KipSigei — APPROVED** [review]

**Review by KipSigei — APPROVED** [review]

**Pull request merged**